pkg/cnab/extensions/dependencies_test.go

```diff
@@ -20,16 +20,63 @@ func TestReadDependencyProperties(t *testing.T) {
 	deps, err := ReadDependencies(*bun)
 
 	assert.NotNil(t, deps, "Dependencies was not populated")
-	assert.Len(t, deps.Requires, 2, "Dependencies.Requires is the wrong length")
-
-	dep := deps.Requires["storage"]
-	assert.NotNil(t, dep, "expected Dependencies.Requires to have an entry for 'storage'")
-	assert.Equal(t, "somecloud/blob-storage", dep.Bundle, "Dependency.Bundle is incorrect")
-	assert.Nil(t, dep.Version, "Dependency.Version should be nil")
-
-	dep = deps.Requires["mysql"]
-	assert.NotNil(t, dep, "expected Dependencies.Requires to have an entry for 'mysql'")
-	assert.Equal(t, "somecloud/mysql", dep.Bundle, "Dependency.Bundle is incorrect")
-	assert.True(t, dep.Version.AllowPrereleases, "Dependency.Bundle.Version.AllowPrereleases should be true")
-	assert.Equal(t, []string{"5.7.x"}, dep.Version.Ranges, "Dependency.Bundle.Version.Ranges is incorrect")
+	assert.Len(t, deps, 2, "Dependencies.Requires is the wrong length")
+
+	for _, dep := range deps {
```

Comment:
Can we not range via `Dependencies.Sequence` here?  (And/or, build in the expectation of a certain order/sequence into these assertions?)

Reply:
If we stop calling ListBySequence inside of ReadDependencies (because it now returns a Dependencies struct), we would start calling it here immediately after ReadDependencies, then could assume the items are ordered without the for loop.

I wouldn't test the ordering much in this function because the very next function below tests ListBySequence in detail. Double testing that function doesn't make much sense IMO and I'd limit the test here to the scope of ReadDependencies.